Simplenote/NoticeController.swift

```diff
@@ -0,0 +1,89 @@
+import UIKit
+
+class NoticeController {
+    static let shared = NoticeController()
+
+    private var notices: [Notice] = []
+    private var current: Notice?
+    private let noticePresenter = NoticePresenter()
+
+    private var activeViewIsBeingTouched: Bool = false
+
+    private init() { }
+
+    func setupNoticeController() {
+        noticePresenter.startListeningToKeyboardNotifications()
+    }
+
+    func present(_ notice: Notice) {
+        if noticePresenter.isPresenting {
+            appendToQueueIfNew(notice)
+            return
+        }
+
+        current = notice
+        let noticeView = makeNoticeView(from: notice)
+
+        noticePresenter.presentNoticeView(noticeView) {
+            if !self.activeViewIsBeingTouched {
+                self.dismiss(noticeView)
+            }
+        }
+    }
+
+    private func dismiss(_ noticeView: NoticeView) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + Times.tapDelay) {
+            self.noticePresenter.dismissNotification {
+                self.current = nil
+                if !self.notices.isEmpty {
+                    self.present(self.notices.removeFirst())
+                }
+            }
+        }
+    }
+
+    private func makeNoticeView(from notice: Notice) -> NoticeView {
+        let noticeView: NoticeView = NoticeView.instantiateFromNib()
+        noticeView.message = notice.message
+        noticeView.actionTitle = notice.action?.title
+        noticeView.handler = notice.action?.handler
```

Comment:
Should it also be dismissed when the user taps on the action?

Reply:
That wasn't in the original design description, but I have wondered if there should be some sort of swipe to dismiss or tap to dismiss the notices.

@SylvesterWilmott WDYT?

Reply:
+1 otherwise the action could be executed N times